black-mulberry-core/src/main/java/com/black/mulberry/core/service/impl/ProductServiceImpl.java

```diff
@@ -0,0 +1,86 @@
+package com.black.mulberry.core.service.impl;
+
+import com.black.mulberry.core.entity.CategoryProduct;
+import com.black.mulberry.core.entity.Product;
+import com.black.mulberry.core.entity.User;
+import com.black.mulberry.core.exception.ProductNotExistException;
+import com.black.mulberry.core.mapper.ProductMapper;
+import com.black.mulberry.core.repository.CategoryProductRepository;
+import com.black.mulberry.core.repository.ProductRepository;
+import com.black.mulberry.core.repository.UserRepository;
+import com.black.mulberry.core.service.ProductService;
+import com.black.mulberry.data.transfer.request.ProductRequest;
+import com.black.mulberry.data.transfer.response.ProductResponse;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+
+@Service
+@RequiredArgsConstructor
+public class ProductServiceImpl implements ProductService {
+    private final ProductRepository productRepository;
+    private final CategoryProductRepository categoryProductRepository;
+    private final UserRepository userRepository;
+    private final ProductMapper productMapper;
+    @Override
+    public ProductResponse save(ProductRequest productRequest) {
+        Optional<User> optionalUser = userRepository.findById(productRequest.getUserId());
+        Optional<CategoryProduct> categoryProduct = categoryProductRepository.findById(productRequest.getCategoryProductId());
+        if(optionalUser.isEmpty() || categoryProduct.isEmpty()){
+           throw new NullPointerException("user or category is null");
```

Comment:
I think NullPointerException is not the place here
throw the appropriate exception NotFoundException
 
it will be more competent if you use not from the repository but from the service methods of services in which there are already the corresponding exceptions when not fi